lib/ansible/galaxy/collection/__init__.py

```diff
@@ -246,6 +265,27 @@ def verify_local_collection(
     return result
 
 
+def verify_file_signature(b_manifest_file, b_detached_signature, keyring):
+    gpg_result, rc = run_gpg_verify(b_manifest_file, b_detached_signature, keyring)
+
+    verify_failed = ""
+    for error in parse_gpg_errors(gpg_result, rc):
+        if not verify_failed:
+            display.vvvv(f"{gpg_result}")
+            signature = to_text(b_detached_signature, errors='surrogate_or_strict')
+            verify_failed += f"Signature verification failed:"
```

Comment:
Looks like this could just be put before the loop

Reply:
Refactored this function and moved this bit before the loop.